include/mqt-core/dd/Operations.hpp

```diff
@@ -34,149 +35,103 @@
-template <class Config>
-qc::MatrixDD
-getStandardOperationDD(const qc::StandardOperation* op, Package<Config>& dd,
-                       const qc::Controls& controls, const qc::Qubit target,
-                       const bool inverse) {
-  GateMatrix gm;
-
-  const auto type = op->getType();
-  const auto& parameter = op->getParameter();
-
-  switch (type) {
-  case qc::I:
-    gm = I_MAT;
-    break;
-  case qc::H:
-    gm = H_MAT;
-    break;
-  case qc::X:
-    gm = X_MAT;
-    break;
-  case qc::Y:
-    gm = Y_MAT;
-    break;
-  case qc::Z:
-    gm = Z_MAT;
-    break;
-  case qc::S:
-    gm = inverse ? SDG_MAT : S_MAT;
-    break;
-  case qc::Sdg:
-    gm = inverse ? S_MAT : SDG_MAT;
-    break;
-  case qc::T:
-    gm = inverse ? TDG_MAT : T_MAT;
-    break;
-  case qc::Tdg:
-    gm = inverse ? T_MAT : TDG_MAT;
-    break;
-  case qc::V:
-    gm = inverse ? VDG_MAT : V_MAT;
-    break;
-  case qc::Vdg:
-    gm = inverse ? V_MAT : VDG_MAT;
-    break;
-  case qc::U:
-    gm = inverse ? uMat(-parameter[1U], -parameter[2U], -parameter[0U])
-                 : uMat(parameter[2U], parameter[1U], parameter[0U]);
-    break;
-  case qc::U2:
-    gm = inverse ? u2Mat(-parameter[0U] + PI, -parameter[1U] - PI)
-                 : u2Mat(parameter[1U], parameter[0U]);
-    break;
-  case qc::P:
-    gm = inverse ? pMat(-parameter[0U]) : pMat(parameter[0U]);
-    break;
-  case qc::SX:
-    gm = inverse ? SXDG_MAT : SX_MAT;
-    break;
-  case qc::SXdg:
-    gm = inverse ? SX_MAT : SXDG_MAT;
-    break;
-  case qc::RX:
-    gm = inverse ? rxMat(-parameter[0U]) : rxMat(parameter[0U]);
-    break;
-  case qc::RY:
-    gm = inverse ? ryMat(-parameter[0U]) : ryMat(parameter[0U]);
-    break;
-  case qc::RZ:
-    gm = inverse ? rzMat(-parameter[0U]) : rzMat(parameter[0U]);
-    break;
-  default:
-    std::ostringstream oss{};
-    oss << "DD for gate" << op->getName() << " not available!";
-    throw qc::QFRException(oss.str());
-  }
-  return dd.makeGateDD(gm, controls, target);
-}
 
-// two-target Operations
 template <class Config>
-qc::MatrixDD
-getStandardOperationDD(const qc::StandardOperation* op, Package<Config>& dd,
-                       const qc::Controls& controls, qc::Qubit target0,
-                       qc::Qubit target1, const bool inverse) {
-  const auto type = op->getType();
-  const auto& parameter = op->getParameter();
-
-  if (type == qc::DCX && inverse) {
-    // DCX is not self-inverse, but the inverse is just swapping the targets
-    std::swap(target0, target1);
+qc::MatrixDD getStandardOperationDD(const qc::StandardOperation& op,
+                                    Package<Config>& dd,
+                                    const qc::Controls& controls,
+                                    const std::vector<qc::Qubit>& targets,
+                                    const bool inverse) {
+
+  auto type = op.getType();
+  std::vector<fp> params = op.getParameter();
+  std::vector<qc::Qubit> targetQubits = targets;
```

Comment:
Yes, good point. I think we can solve this without a separate function and just have it inline like this:
```c++
if (not inverse) {
   // return gate directly
}
// construct inverse gate
std::vector<qc::Qubit> targetQubits = targets;
switch {
  // ...
}
// return inverse gate
```
This would have a duplicate call to gate creations, but as this is only two lines, I think this is ok.

Reply:
That works for me 👍🏼 sounds like a nice compromise.